sdv/datasets/demo.py

```diff
@@ -0,0 +1,140 @@
+"""Methods to load datasets."""
+
+import io
+import logging
+import os
+import shutil
+import tempfile
+import urllib.request
+from zipfile import ZipFile
+
+import pandas as pd
+
+from sdv.datasets.errors import InvalidArgumentError
+from sdv.metadata.multi_table import MultiTableMetadata
+from sdv.metadata.single_table import SingleTableMetadata
+
+LOGGER = logging.getLogger(__name__)
+BUCKET_URL = 'https://sdv-demo-datasets.s3.amazonaws.com'
+
+
+def _get_dataset_url(modality, dataset_name):
+    return os.path.join(BUCKET_URL, modality.upper(), dataset_name + '.zip')
+
+
+def _validate_args_download_demo(modality, dataset_name, output_folder_name):
+    possible_modalities = ['single_table', 'multi_table', 'sequential']
+    if modality not in possible_modalities:
+        raise InvalidArgumentError(f"'modality' must be in {possible_modalities}.")
+
+    if output_folder_name and os.path.exists(output_folder_name):
+        raise InvalidArgumentError(
+            f"Folder '{output_folder_name}' already exists. Please specify a different name "
+            "or use 'load_csvs' to load from an existing folder."
+        )
+
+    dataset_url = _get_dataset_url(modality, dataset_name)
+    try:
+        urllib.request.urlopen(dataset_url)
+    except urllib.error.HTTPError:
+        # If the dataset exists in the wrong modality, raise an error
+        other_modalities = set(possible_modalities) - {modality}
+        for other_modality in other_modalities:
+            dataset_url = _get_dataset_url(other_modality, dataset_name)
+            try:
+                urllib.request.urlopen(dataset_url)
+                raise InvalidArgumentError(
+                    f"Dataset name '{dataset_name}' is a '{other_modality}' dataset. "
+                    f"Use 'load_{other_modality}_demo' to load this dataset."
+                )
+            except urllib.error.HTTPError:
+                pass
+
+        # If the dataset doesn't exist at all, raise different error
+        raise InvalidArgumentError(
+            f"Invalid dataset name '{dataset_name}'. "
+            "Use 'list_available_demos' to get a list of demo datasets."
+        )
```

Comment:
I think we should get rid of this whole section. Instead, during the actual `_download` method, we should throw that in a try catch and raise an error if it can't be downloaded suggesting that possibly it is a different modality.

Reply:
Yeah, I implemented like this to validate whether the dataset is in another modality. Since we removed it, this is unnecessary.

PS: also, I originally implemented it using the `requests` library, so I could make a HEAD request, which doesn't cost much. But I forgot that we don't have `requests`, and I don't know how to do it with `urllib`...